src/main/java/uk/nhs/hee/trainee/details/api/ProgrammeMembershipResource.java

```diff
@@ -108,13 +122,29 @@ public ResponseEntity<Boolean> deleteProgrammeMemberships(
    */
   @PostMapping("/{programmeMembershipId}/sign-coj")
   public ResponseEntity<ProgrammeMembershipDto> signCoj(
-      @PathVariable(name = "programmeMembershipId") String programmeMembershipId) {
+      @PathVariable String programmeMembershipId,
+      @RequestHeader(HttpHeaders.AUTHORIZATION) String token) {
     log.info("Signing COJ with Programme Membership ID {}", programmeMembershipId);
+
+    String[] tokenSections = token.split("\\.");
+    byte[] payloadBytes = Base64.getUrlDecoder()
+        .decode(tokenSections[1].getBytes(StandardCharsets.UTF_8));
+    String traineeTisId;
+
+    try {
+      Map<?, ?> payload = objectMapper.readValue(payloadBytes, Map.class);
+      traineeTisId = (String) payload.get(TIS_ID_ATTRIBUTE);
+    } catch (IOException e) {
```

Comment:
Maybe move this to a Util class to avoid duplication with `TraineeProfileResource`

We do the same in [tis-trainee-forms](https://github.com/Health-Education-England/tis-trainee-forms/blob/7c5f96f76cbc8d4e05be969450c55e4a0273649a/src/main/java/uk/nhs/hee/tis/trainee/forms/api/util/AuthTokenUtil.java#L52)